itchiodl/library.py

```diff
@@ -40,39 +41,67 @@ def load_owned_games(self):
                 break
             page += 1
 
+    def load_game_page_keys(self, page):
+        """Load a page of game keys via the API"""
+        duplicates = 0
+        print("Loading page", page)
+        r = requests.get(
+            f"https://api.itch.io/profile/owned-keys?page={page}",
+            headers={"Authorization": self.login},
+        )
+        j = json.loads(r.text)
+
+        for s in j["owned_keys"]:
+            if str(s.get("game_id")) in self.key_pairs:
+                duplicates+=1
+                # The duplicate download key check assumes that a consecutive
+                # string of ten identical game ids indicates the list has not changed
+                # Duplicate game keys can happen if you buy multiple bundles containing the same item
+                # print("duplicate game ID", s["game_id"]) # Old debug line
+                if duplicates > 9 and exists("key_pairs.json"):
+                    print("Assuming that the owned keys have not changed")
+                    return 0
+            else:
+                self.key_pairs.update({str(s["game_id"]):s["id"]})
+                duplicates = 0
+
+        return len(j["owned_keys"])
+
+    def load_owned_keys(self):
+        """Load game_id:download_id pairs only and stores them in the library.keyPairs dictionary"""
+        page = 1
+        if exists("key_pairs.json"):
+            infile = open("key_pairs.json","r")
+            self.key_pairs.update(json.loads(infile.read()))
+            infile.close()
+        while True:
+            n = self.load_game_page_keys(page)
+            if n == 0:
+                break
+            page += 1
+
+        outfile = open("key_pairs.json", "w")
```

Comment:
should make use of `with open(...) as file:` but should use pathlib overall

Reply:
Converting this line to pathlib